core/blockchain.go

```diff
@@ -2908,12 +2918,14 @@ func (bc *BlockChain) addDelegationIndex(
 	// If there is an existing delegation, just return
 	validatorAddressBytes := validatorAddress.Bytes()
 	for _, delegation := range delegations {
+		// this will happen when someone delegates more to the same validator
+		// since a delegationIndex only stores the blockNum, validator address and block number
```

Comment:
stores the index in the delegation list.

Reply:
Will update the comment but my intention was to say that it doesn't store the `Amount`, and it is only representative of the Index in the delegation list.